File: backend/deploy-service/src/main/java/com/zerocode/deploy/vo/ApiResponse.java
package com.zerocode.deploy.vo;

public record ApiResponse<T>(
        int code,
        T data,
        String message) {
    public static <T> ApiResponse<T> ok(T data) {
        return new ApiResponse<>(200, data, "ok");
    }

    public static <T> ApiResponse<T> error(int code, String message) {
        return new ApiResponse<>(code, null, message);
    }
}

File: ai-services/ai-orchestrator/app/models/api_response.py
from typing import Any


def ok(data: Any) -> dict[str, Any]:
    return {
        "code": 0,
        "data": data,
        "message": "ok",
    }


def fail(code: int, message: str) -> dict[str, Any]:
    return {
        "code": code,
        "data": None,
        "message": message,
    }
